bentoml/yatai/__init__.py

```diff
@@ -75,4 +83,80 @@ def get_yatai_service(
+    track_server('yatai-service-grpc-server')
+    yatai_service = get_yatai_service(db_url=db_url, repo_base_url=repo_base_url)
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
+    add_YataiServicer_to_server(yatai_service, server)
+    debug_mode = config().getboolean('core', 'debug')
+    if debug_mode:
+        try:
+            logger.debug('Enabling gRPC server reflection for debugging')
+            from grpc_reflection.v1alpha import reflection
+            from bentoml.proto import yatai_service_pb2
+
+            SERVICE_NAMES = (
+                yatai_service_pb2.DESCRIPTOR.services_by_name['Yatai'].full_name,
+                reflection.SERVICE_NAME,
+            )
+            reflection.enable_server_reflection(SERVICE_NAMES, server)
+        except ImportError:
+            logger.debug(
+                'Failed enabling gRPC server reflection, missing required package: '
+                '"pip install grpcio-reflection"'
+            )
+    server.add_insecure_port(f'[::]:{grpc_port}')
+    server.start()
+    if with_ui:
+        yatai_grpc_server_addess = f'localhost:{grpc_port}'
+        proc = async_start_yatai_service_web_ui(
+            yatai_grpc_server_addess, ui_port, debug_mode
+        )
+    logger.info(
+        f'* Starting BentoML YataiService gRPC Server\n'
+        f'* Debug mode: { "on" if debug_mode else "off"}\n'
+        f'* Web UI: {f"running on http://127.0.0.1:{ui_port}" if with_ui else "off"}\n'
+        f'* Running on 127.0.0.1:{grpc_port} (Press CTRL+C to quit)\n'
+        f'* Usage: `bentoml config set yatai_service.url=127.0.0.1:{grpc_port}`\n'
+        f'* Help and instructions: '
+        f'https://docs.bentoml.org/en/latest/concepts/yatai_service.html'
+    )
+    try:
+        while True:
+            time.sleep(_ONE_DAY_IN_SECONDS)
+    except KeyboardInterrupt:
+        logger.info("Terminating YataiService gRPC server..")
+        if with_ui and proc:
+            proc.terminate()
+        server.stop(grace=None)
+
+
+def async_start_yatai_service_web_ui(yatai_server_address, ui_port, debug_mode):
+    if ui_port is not None:
+        ui_port = ui_port if isinstance(ui_port, str) else str(ui_port)
+    web_ui_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), 'web'))
+    if debug_mode:
+        # WIP need to find way to include port and yatai server address into watch
+        web_ui_command = ['npm', 'watch']
+    else:
+        # NOTE, we need to make sure we build dist before we start this
+        if not os.path.exists(os.path.join(web_ui_dir, 'dist')):
+            build_web_dist = subprocess.Popen(
+                ['npm', 'build'],
+                cwd=web_ui_dir,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE
+            )
+            logger.debug(build_web_dist.stdout.read().decode('utf-8'))
+        web_ui_command = ['node', 'dist/index.js', yatai_server_address, ui_port]
```

Comment:
should we make sure to restart the node process when the web UI server crashes? would something like https://github.com/foreversd/forever work? or maybe something we can do from the python side?

Reply:
e.g. with psutil library? https://psutil.readthedocs.io/en/latest/

Reply:
I think we can use forever to keep it running.